package/src/components/ProgressControl/ProgressControl.tsx

```diff
@@ -109,28 +145,27 @@ export const ProgressControl = React.memo(
         <Animated.View style={[styles.filledStyle, animatedStyles, filledStyles]} />
         <GestureDetector gesture={pan}>
           <Animated.View style={[thumbStyles, thumb]}>
-            {onProgressDrag ? <ProgressControlThumb /> : null}
+            {onEndDrag || onProgressDrag ? <ProgressControlThumb /> : null}
           </Animated.View>
         </GestureDetector>
       </View>
     );
   },
   (prevProps, nextProps) => {
-    if (
-      prevProps.duration === nextProps.duration &&
-      prevProps.progress === nextProps.progress &&
-      prevProps.width === nextProps.width
-    )
-      return true;
-    else return false;
+    if (prevProps.duration !== nextProps.duration) return false;
```

Comment:
What's the idea behind this change ? Can't we just:

```
return (prevProps.duration === nextProps.duration &&
      prevProps.progress === nextProps.progress &&
      prevProps.width === nextProps.width)
```
here ?

Reply:
Yes, we can. I just added it since I felt it would do early exits.

Reply:
That's a good point, but early exits would matter if we for example did something like:

```
prevProps.someArray.some(item => nextProps.someArray.includes(item))
```
or something else in `O(n)` time - these are all `O(1)` so it's probably better to have the cleaner and more readable version.

Reply:
I will revert it 👍🏻

Reply:
You can actually change it to:

```
return (prevProps.duration === nextProps.duration &&
      prevProps.progress === nextProps.progress &&
      prevProps.width === nextProps.width)
```
I'm pretty sure, unless you see some issue with that

Reply:
Yes, pushed it as the one you suggested